llvm/lib/Target/DirectX/DXILPrettyPrinter.cpp

```diff
@@ -155,8 +155,8 @@ struct FormatResourceDimension
     case dxil::ResourceKind::StructuredBuffer:
       if (!Item.isUAV())
         OS << "r/o";
-      else if (Item.getUAV().HasCounter)
-        OS << "r/w+cnt";
+      //      else if (Item.getUAV().HasCounter)
+      //        OS << "r/w+cnt";
```

Comment:
Not sure what to do here. `Item` is `ResourceTypeInfo` and there is no `ResourceType` to be seen so idk how to replicate this exact functionality.

Maybe it can just be deleted? But I wanted input first. @hekota @bogner

Reply:
`FormatResourceDimension` is used from [here](https://github.com/llvm/llvm-project/blob/1cec5fffd8fddd9d85b516f876093b0e3f0eec5f/llvm/lib/Target/DirectX/DXILPrettyPrinter.cpp#L241) where `ResourceType` is available, so can add the `ResourceType` as an additional argument to `FormatResourceDimension`.